lib/em-http/http_connection_options.rb

```diff
@@ -42,4 +42,19 @@ def connect_proxy?
   def socks_proxy?
     @proxy && (@proxy[:type] == :socks5)
   end
+
+  def proxy
+    @proxy ||= proxy_env_uri && { :host => proxy_env_uri.host, :port => proxy_env_uri.port, :type => :http }
```

Comment:
I'm wondering if we can unroll and make this a bit more explicit..

- If we're parsing out the ENV values, we should reflect the values in `options[:proxy]`
- We should clearly define precedence.. I would assume that manually passing in proxy settings in options hash would take precedence over the ENV variables; if options are not present then default to ENV variables.

Roughly, I'm picturing...

```
 @proxy = options[:proxy] || env_proxy

  ...

  def env_proxy
    proxy_str = if @https
        ENV['HTTPS_PROXY'] || ENV['https_proxy'] 
     else 
        ENV['HTTP_PROXY'] || ENV['http_proxy']
     end

     proxy_uri = Addressable::URI::parse(proxy_str || ENV['ALL_PROXY'])
     {:host => proxy_uri.host, :port => proxy_uri.port, type => :http}
  end
```

WDYT?